File: 2833.furthest-point-from-origin.py
#
# @lc app=leetcode id=2833 lang=python
#
# [2833] Furthest Point From Origin
#

# @lc code=start
class Solution(object):
    def furthestDistanceFromOrigin(self, moves):
        """
        :type moves: str
        :rtype: int
        """
        a = 0
        b = 0
        for i in moves:
            if i =='L':
                a += 1
            elif i == 'R':
                a -= 1
            else:
                b += 1
        return abs(a) + b
        
# @lc code=end

File: 2833.furthest-point-from-origin.java
/*
 * @lc app=leetcode id=2833 lang=java
 *
 * [2833] Furthest Point From Origin
 */

// @lc code=start
class Solution {
    public int furthestDistanceFromOrigin(String moves) {
        int a = 0;
    int b = 0;
    for (char c : moves.toCharArray()) {
      if (c == 'L') {
        a++;
      }
      else if (c == 'R') {
        a--;
      }
      else {
        b++;
      }
    }
    return Math.abs(a) + b;
    }
}
// @lc code=end

